cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/cmd/PullCommand.java

```diff
@@ -81,28 +83,34 @@ public void execute() {
 
         String resourceName = argList.get(0);
         String orgName;
+        String packageName;
         String moduleName;
         String version;
 
         // Get org-name
-        int orgNameIndex = resourceName.indexOf("/");
-        if (orgNameIndex != -1) {
-            orgName = resourceName.substring(0, orgNameIndex);
-            if (orgName.equals("ballerina")) {
-                throw LauncherUtils.createLauncherException("`Ballerina` is the builtin organization and its modules"
-                                                                    + " are included in the runtime.");
-            }
-        } else {
-            throw LauncherUtils.createLauncherException("no module-name provided");
+        if (!validateModuleName(resourceName)) {
+            CommandUtil.printError(outStream,
+                    "invalid module name. Provide the module-name with the org-name",
+                    "ballerina pull {<org-name>/<module-name> | <org-name>/<module-name>:<version>}",
+                    false);
+            Runtime.getRuntime().exit(1);
+            return;
+        }
+
+        String[] module = resourceName.split("/");
+        orgName = module[0];
+        packageName = module[1];
+        if (orgName.equals("ballerina")) {
+            throw LauncherUtils.createLauncherException("`Ballerina` is the builtin organization and its modules"
+                    + " are included in the runtime.");
         }
 
         // Get module name
-        int packageNameIndex = resourceName.indexOf(":");
-        if (packageNameIndex != -1) { // version is provided
-            moduleName = resourceName.substring(orgNameIndex + 1, packageNameIndex);
-            version = resourceName.substring(packageNameIndex + 1, resourceName.length());
+        if (packageName.contains(":")) { // version is provided
```

Comment:
Shall we use one spilt function and use return array to get module name and version? we can remove contains and check split array